packages/typescript/index.js

```diff
@@ -29,6 +29,22 @@ function typescript (options) {
         }
       ]
     },
-    plugins: options ? [new CheckerPlugin(), new TsConfigPathsPlugin({ tsconfig: options.tsconfig, compiler: options.compiler })] : [new CheckerPlugin()]
+    plugins: [
+      new CheckerPlugin()
+    ].concat(
+      options ? [
+        new TsConfigPathsPlugin({ tsconfig: options.tsconfig, compiler: options.compiler })
+      ] : []
+    )
   })
+  
+  return Object.assign(setter, { pre })
+}
+
+function pre (context) {
+  if ('application/x-typescript' in context.fileType.all()) {
```

Comment:
I would not leave an empty if branch, I would use:
`if (!('application/x-typescript' in context.fileType.all())) {`
and just have the else branch